### PR TITLE
Sentence onset improvements

### PR DESCRIPTION
**Pull request by ss8319**
-Remove the conversion of time from ms to seconds 
-Added more detailed logging of more metrics- Specificity (Rate of detection of Positive Samples) and Sensitivity (Rate of detection of Negative Samples), Precision
-Added weighted BCE to better deal with imbalanced classes

### CODE REVIEW

File: configs/neural_conv_decoder/neural_conv_decoder_sentence_onset.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ data_params:
     negatives_per_positive: 5 
     #Negative examples are sampled within each sentence, at least `negative_margin_s` seconds after onset
     negative_margin_s: 0.75 #
-    sentence_csv_path: data/stimuli/all_sentences_podcast.csv
+    sentence_csv_path: data/stimuli/all_sentences_podcast_cleaned.csv
```
Comment (zparis99) [inline configs/neural_conv_decoder/neural_conv_decoder_sentence_onset.yml]: Can we add this to the repo maybe so it's easily ported with the code?

File: metrics.py [hunk 2]
```diff
@@ -122,6 +169,117 @@ def f1_binary(pred: torch.Tensor, true: torch.Tensor) -> float:
...
+    """
+    # Ensure 1D
+    if pred.ndim > 1:
+        pred = pred.squeeze(-1)
+    if true.ndim > 1:
+        true = true.squeeze(-1)
+
+    y_true = true.detach().cpu().numpy().astype(int)
+    # Convert to probabilities if needed; assume in [0,1] otherwise
+    if pred.detach().min() < 0 or pred.detach().max() > 1:
```
Comment (zparis99) [inline metrics.py]: Same thought here. I'd rather just have a well-defined API about whether it should be logits or probabilities rather than potentially changing results silently.

File: metrics.py [hunk 3]
```diff
@@ -122,6 +169,117 @@ def f1_binary(pred: torch.Tensor, true: torch.Tensor) -> float:
...
```
Comment (zparis99) [inline metrics.py]: Same here.

File: task_utils.py [hunk 4]
```diff
@@ -30,15 +30,10 @@ def sentence_onset_task(data_params: DataParams):
     # Resolve CSV path
     default_csv = os.path.join(data_params.data_root, "all_sentences_podcast.csv")
     csv_path = tp.get("sentence_csv_path", default_csv)
+    print(csv_path)
```
Comment (zparis99) [inline task_utils.py]: Leftover from debug?

**Comment by ss8319** [comment]
@zparis99 the different issues should be resolved now

Thanks for the review, honestly I should have picked up the stuff you mentioned in the first place.
